corfudb-tools/src/main/java/org/corfudb/browser/CorfuStoreBrowser.java

```diff
@@ -252,20 +263,107 @@ public int printAllProtoDescriptors() {
         log.info("\n======================\n");
         String fullName = TableRegistry.getFullyQualifiedTableName(namespace, tablename);
         UUID streamUUID = UUID.nameUUIDFromBytes(fullName.getBytes());
-        CorfuTable<CorfuDynamicKey, CorfuDynamicRecord> table = getTable(namespace, tablename);
+        runtime.getObjectsView().TXBegin();
+        CorfuTable<CorfuDynamicKey, CorfuDynamicRecord> table =
+            getTable(namespace, tablename);
         int tableSize = table.size();
         log.info("Table {} in namespace {} with ID {} with {} entries will be dropped...",
                 tablename, namespace, streamUUID.toString(), tableSize);
         table.clear();
+        runtime.getObjectsView().TXEnd();
         log.info("Table cleared successfully");
         log.info("\n======================\n");
         return tableSize;
     }
 
+    /**
+     * Edit a record in a table and namespace
+     * @param namespace namespace of the table
+     * @param tableName name of the table
+     * @param keyToEdit JSON string representing the key whose corresponding
+     *  record is to be editted
+     * @param newRecord JSON string representing the new value to be inserted
+     *  against keyToEdit
+     * @return CorfuDynamicRecord the edited CorfuDynamicRecord.  null if no
+     *  record was edited, either due to an error or key not found.
+     */
+    public CorfuDynamicRecord editRecord(String namespace, String tableName,
+        String keyToEdit, String newRecord) {
+        verifyNamespaceAndTableName(namespace, tableName, true);
+
+        if (keyToEdit == null) {
+            throw new IllegalArgumentException("Please specify the key to " +
+                "edit --keyToEdit");
+        }
+
+        if (newRecord == null) {
+            throw new IllegalArgumentException("Please specify the newRecord " +
+                "--newRecord");
+        }
+
+        log.info("\n======================\n");
+        String fullName = TableRegistry.getFullyQualifiedTableName(namespace,
+            tableName);
+        UUID streamUUID = CorfuRuntime.getStreamID(fullName);
+
+        TableName tableNameProto = TableName.newBuilder().setTableName(tableName)
+            .setNamespace(namespace).build();
+
+        CorfuStoreMetadata.KeyValueDefaultRecord keyValueDefaultRecord =
+            dynamicProtobufSerializer.getCachedKeyValueTypesTable().get(tableNameProto);
+
+        Any defaultKeyAny = keyValueDefaultRecord.getKey();
+        Any defaultValueAny = keyValueDefaultRecord.getValue();
+        DynamicMessage keyMsg =
+            dynamicProtobufSerializer.createDynamicMessageFromJson(defaultKeyAny,
+                keyToEdit);
+        DynamicMessage newValueMsg =
+            dynamicProtobufSerializer.createDynamicMessageFromJson(defaultValueAny,
+            newRecord);
+
+        if (keyMsg == null || newValueMsg == null) {
+            return null;
+        }
+
+        runtime.getObjectsView().TXBegin();
```

Comment:
Could the TX begin later? how about begin at line 339?

Reply:
this is a read-modify-write operation.  So getTable() must be done from within the transaction.  But I have moved it to later.